blog/executor-plugins.md

```diff
@@ -0,0 +1,348 @@
+
+   - The **`Metadata`** method returns basic information about your plugin. This data is used when the plugin index is [generated in an automated way](https://docs.botkube.io/plugin/repo).
+
+   - The **`Execute`** method is the heart of your executor plugin. This method runs your business logic and returns the output as plaintext. Next, the Botkube core sends back the response to a given communication platform.
+
+2. To download the imported dependencies, in your terminal, run:
+
+   ```bash
+   go mod tidy
+   ```
+
+3. Great! At this stage you already have a functional Botkube executor plugin. However, for now, it only responds with a hard-coded "Aloha!" greeting. But it can do that already on all supported communication platforms.
+
+   ![](./assets/demo-gh-aloha.gif)
+
+   Don't worry, in the next steps, we will implement our business logic.
+
+4. Add support for user configuration:
+
+   <Gist id="39fc1614e2849319314786fc53efe52d" />
+
+   For each `Execute` method call, Botkube attaches the list of associated configurations. The input parameters are defined by the user, when enabling a given plugin:
+
+   ```yaml
+   executors:
+     "plugin-based":
+       repo-name/gh:
+         enabled: true # If not enabled, plugin is not downloaded and started.
+         config: # Plugin's specific configuration.
+           github:
+             repository: "mszostok/repository"
+             token: "github_pat_foo"
+             issueTemplate: |
+               ## Description
+
+               This issue refers to the problems connected with {{ .Type | code "bash" }} in namespace {{ .Namespace | code "bash" }}
+
+               {{ .Logs | code "bash"}}
+   ```
+
+   In our case, we need to have a GitHub token, GitHub repository where the issue should be created, and an issue template. The remaining parameters can be hard-coded on the plugin side, however, your plugin will be more flexible if you allow your users to change it without rebuilding your plugins.
+
+   It's up to the plugin author to merge the passed configurations. You can use our helper function from the plugin extension package (`pluginx`). To learn more, see [Passing configuration to your plugin](https://docs.botkube.io/plugin/custom-executor#passing-configuration-to-your-plugin).
+
+5. Let's implement command parsing to properly understand command syntax:
+
+   Our goal is to parse `gh create issue KIND/NAME [-n, --namespace]`.
+
+   There are a lot of great libraries supporting command parsing. The most popular is probably [`cobra`](https://github.com/spf13/cobra), but for our use case, we will just use the helper function from our plugin extension package.
+
+   <Gist id="6965d521ebb04a858b04df60d04c8af6" />
+
+   Under the hood, the `pluginx.ParseCommand` method uses [`go-arg`](https://github.com/alexflint/go-arg).
+
+6. We are almost there! Now let's fetch the issue details:
+
+   <Gist id="85896e45d9d074143d60343b1d2f941c" />
```

Comment:
Update: I had a problem where I got no issue url and the GH issues weren't created. So that's why it would be crucial to report such errors to know what's wrong 👍  


After copying all the snippets I got:
```
Preview
APP  [11:14 AM](https://temptestworkspaceco.slack.com/archives/C03US4YU83G/p1674728087865149)
gh create issue job/oops -n default on executor-how-to
New issue created successfully! 🎉

Issue URL:
```

without any errors as they were ignored.

When I copied the code from repo template it worked instantly, so I think it could be something about creating GH issue as this part is different: https://github.com/pkosiec/botkube-plugins/commit/776889a9fe90b3b1d6894e27c253106d32fb9816

Anyway, please add a very basic error handling 🙏

Reply:
>We could use the approach with exitOnError function that does log.Fatal without wrapping the error. WDYT?

Unfortunately that will be the worst ad the plugin won't be restarted and user will also not see the error at all. If you look on the revisions on each gist you will find that there were error handling, but I decided to remove it as it increases the readability. It's common in blogpost that you skip error handling, to focus on the main path. WDYT about adding a note about that the error handling is skipped and if needed to take a look on the example repo. 


--

Yes, I updated helpers, but blog post not. I will do that once helpers will be finally merged 👍

Reply:
> WDYT about adding a note about that the error handling is skipped and if needed to take a look on the example repo.

Sounds good to me to have such note 👍 Hope Webflow supports that 🤔